src/pages/instances/InstanceList.tsx

```diff
@@ -279,10 +277,10 @@ const InstanceList: FC = () => {
     pageData: pageInstances,
```

Comment:
I wonder if it would be better to not unpack here: `const page = usePagination...`. Then we could simplify below to `<Pagination {...page}` and adjust the 3-4 usages of details from it to calling `page.pageData`. But we'd need to rename `setCurrentPage` to `paginate` in the `Pagination` component to make the props match.

Reply:
I would avoid this change, because I don't see any particular benefit and it complicates referencing `pageInstances` - e.g. as the `rows` prop of the `MainTable`, or for the check in `visibleCount`.
For example, you'd need to do something like this:
```
<MainTable
  headers={getHeaders(userHidden.concat(sizeHidden))}
  rows={page.pageData}
```
And similar for the `visibleCount`. Seems to me that we're complicating things instead of making them simpler and cleaner.